infrastructure/oss/vulnerability_count_test.go

```diff
@@ -0,0 +1,193 @@
+		assert.Equal(t, filePath, vci.FilePath)
+		assert.Equal(t, r, vci.Range)
+		assert.Equal(t, 4, vci.total)
+		assert.Equal(t, 1, vci.severityCounts[snyk.High])
+		assert.Equal(t, 3, vci.severityCounts[snyk.Medium])
+		assert.Equal(t, "id1", vci.mostSevereVulnerabilityId)
+		return nil
+	}
+
+	err := vc.ProcessVulnerabilityCount(issues, testFunc)
+
+	assert.NoError(t, err)
+}
+
+func TestVulnerabilityCountImpl_ProcessVulnerabilityCount_GroupByFile(t *testing.T) {
+	c := testutil.UnitTest(t)
+	r := snyk.Range{
+		Start: snyk.Position{
+			Line: 10,
+		},
+		End: snyk.Position{
+			Line:      10,
+			Character: math.MaxInt32,
+		},
+	}
+	filePath := "/test/filePath"
+	filePath2 := "/test/filePath2"
+	issues := []snyk.Issue{
+		{
+			ID:               "id1",
+			Severity:         snyk.High,
+			AffectedFilePath: filePath,
+			Range:            r,
+		},
+		{
+			ID:               "id2",
+			Severity:         snyk.Medium,
+			AffectedFilePath: filePath2,
+			Range:            r,
+		},
+	}
+
+	vc := NewVulnerabilityCountService(c)
+	testFunc := func(vci VulnerabilityCountInformation) error {
+		if vci.FilePath == filePath {
+			assert.Equal(t, 1, vci.total)
+			assert.Equal(t, 1, vci.severityCounts[snyk.High])
+			assert.Equal(t, "id1", vci.mostSevereVulnerabilityId)
+		} else if vci.FilePath == filePath2 {
+			assert.Equal(t, 1, vci.total)
+			assert.Equal(t, 1, vci.severityCounts[snyk.Medium])
+			assert.Equal(t, "id2", vci.mostSevereVulnerabilityId)
+		} else {
+			assert.Fail(t, "Unexpected file path")
+		}
+		return nil
+	}
+
+	err := vc.ProcessVulnerabilityCount(issues, testFunc)
+
+	assert.NoError(t, err)
+}
+
+func TestVulnerabilityCountImpl_ProcessVulnerabilityCount_GroupByRange(t *testing.T) {
+	c := testutil.UnitTest(t)
+	range1 := snyk.Range{
+		Start: snyk.Position{
+			Line: 10,
+		},
+		End: snyk.Position{
+			Line:      10,
+			Character: math.MaxInt32,
+		},
+	}
+	range2 := snyk.Range{
+		Start: snyk.Position{
+			Line: 11,
+		},
+		End: snyk.Position{
+			Line:      11,
+			Character: math.MaxInt32,
+		},
+	}
+	filePath := "/test/filePath"
+	issues := []snyk.Issue{
+		{
+			ID:               "id1",
+			Severity:         snyk.High,
+			AffectedFilePath: filePath,
+			Range:            range1,
+		},
+		{
+			ID:               "id2",
+			Severity:         snyk.Medium,
+			AffectedFilePath: filePath,
+			Range:            range2,
+		},
+	}
+
+	vc := NewVulnerabilityCountService(c)
+	testFunc := func(vci VulnerabilityCountInformation) error {
+		if vci.Range == range1 {
+			assert.Equal(t, 1, vci.total)
+			assert.Equal(t, 1, vci.severityCounts[snyk.High])
+			assert.Equal(t, "id1", vci.mostSevereVulnerabilityId)
+		} else if vci.Range == range2 {
+			assert.Equal(t, 1, vci.total)
+			assert.Equal(t, 1, vci.severityCounts[snyk.Medium])
+			assert.Equal(t, "id2", vci.mostSevereVulnerabilityId)
+		} else {
+			assert.Fail(t, "Unexpected file path")
+		}
+		return nil
+	}
+
+	err := vc.ProcessVulnerabilityCount(issues, testFunc)
```

Comment:
Suggestion: Currently these tests would all pass even if the processor would not be called at all. How about setting an expectation of how often a processor is called in a test.

Reply:
True. I'll add it.

Reply:
Will add a channel post for each invocation, read from it from the test and increase the count on read, then check the count.